src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1182,6 +1208,59 @@ export class StudyViewPageStore
                             this.molecularProfileMapByType,
                             selectedSamples
                         );
+                    } else if (chartMeta.dataType === 'Custom_Data') {
+                        const clinicalAttribute = chartMeta.clinicalAttribute;
+                        const sampleKeyToData = this._customChartsSelectedCases.get(
+                            chartMeta.uniqueKey
+                        );
+                        if (clinicalAttribute) {
+                            if (selectedSamples.length === 0) {
+                                data = [];
+                            } else {
+                                if (sampleKeyToData !== undefined) {
+                                    const sampleInformation = sampleKeyToData.map(
+                                        item => ({
+                                            sampleId: item.sampleId,
+                                            studyId: item.studyId,
+                                            patientId: item.patientId,
+                                            value: item.value,
+                                        })
+                                    );
+                                    // Combine selectedSamples with sampleKeyToData to get the desired data object
+                                    let merged = [];
+                                    for (
+                                        let i = 0;
+                                        i < selectedSamples.length;
+                                        i++
+                                    ) {
+                                        merged.push({
+                                            ...selectedSamples[i],
+                                            ...sampleInformation.find(
+                                                itmInner =>
+                                                    itmInner.sampleId ===
+                                                    selectedSamples[i].sampleId
+                                            ),
+                                        });
+                                    }
+
+                                    const inputObject: any[] = merged.filter(
+                                        item =>
+                                            item.uniqueSampleKey !== undefined
+                                    );
+                                    const outputObject: any[] = inputObject.map(
```

Comment:
also, we should almost never use the `any` type in our logic.

Reply:
We can define a new type `comparisonCustomData` with the following code to generate the selectedSampleData:
```
const selectedSampleData: comparisonCustomData[] = 
sampleInformation.map(
              sample =>
                ({ ...sample,                                       
                    ...selectedSamples.find(                                                   
                                           itmInner =>                                                       
                                           itmInner.sampleId ===                                                        
                                           sample.sampleId),
                } as comparisonCustomData)
```
and:
```
data = selectedSampleData.map(item => {
                     return {
                            clinicalAttribute: clinicalAttribute,
                            clinicalAttributeId:
                                              clinicalAttribute.clinicalAttributeId,
                            ...item,
                                        } as ClinicalData;
                                    })
```